Postgres LargeObject reactive streaming action

enragedginger opened the pull request:
Hello,

I was recently working on a side project using Slick, Postgres, and Play and needed to stream LargeObject instances directly from a Postgres database down to a client. After a large amount of reading and scouring of the Internet, I decided to roll my own action for doing this. It allows you to open an InputStream on a Postgres LargeObject and stream it down to a client. Basically, running the action with `database.stream` produces a `DatabasePublisher[Array[Byte]]` instance. From the controller layer of my application, I just do this to open a chunked reactive stream down to the client
```
Ok.chunked(Source.fromPublisher(someFunctionThatProducesTheDatabasePublisher())).
  as("some/mime-type").
  withHeaders(
    "Content-Disposition" ->"attachment; filename=some-file.txt"
)
```

So, at this point, I'm looking for guidance and feedback.

1. Is this something that belongs in slick-pg? It looks like it's the first piece of functionality in this project to extend `DBIOAction`, but it's heavily dependent on Postgres, so I figured it was better to put it here than it was to put it back into Slick directly.
2. Where does this belong in the project? Should it be its own add-on or is the current location fine?
3. I plan on adding documentation (definitely), tests (potentially), and (probably) another action for importing LargeObjects into the system. What are your thoughts on this?

Thanks for such a great project!

tminglei commented:
@enragedginger yes, it's better to be added to slick-pg instead of slick.
And, your current location is fine, pls go ahead.

_Thanks! :+1:_

tminglei commented on README.md:
```diff
@@ -17,6 +17,7 @@ Slick-pg
 - Inet/MacAddr
 - `text` Search
 - `postgis` Geometry
+- Large Object
```
I'd like to put it under 'Currently supported pg features' instead of here, because when we said it's a pg data type, we mean its corresponding scala type and type related operator/functions.

tminglei commented on README.md:
````diff
@@ -129,7 +130,31 @@ object tests extends TableQuery(new TestTable(_)) {
+  withHeaders(
+    "Content-Disposition" -> "attachment; filename=somefilename.txt"
+  )
+```
````
To keep the main README clear and clean, it's better to put it to its source comments, its own README and its test cases.

enragedginger commented:
Okay, I've migrated the documentation to a README.md in the `lobj` package and added an action for uploading data from an InputStream to the Large Object store and returning the `oid` of the object. I also added a test case for uploading and downloading a Large Object.

This is ready for review.

tminglei commented:
Merged. _Thanks! :+1:_

bennythomps commented:
👍
